### ISSUE TITLE
Error with installing GenomicConsensus (python-related)

### ISSUE THREAD

**Issue by terrymun** [PacificBiosciences/pitchfork#332]
Thanks for the amazing documentation. I've managed to get a few packages built without much issue, but ran into a problem when attempting to install GenomicConsensus.

I have installed pitchfork in my own `bin` directory. When running just `make GenomicConsensus` without specifying the python path via `HAVE_PYTHON = /path/to/python`, it complains that it cannot retrieve PySam over https, which is expected (the computing cluster that I am working with bans https connections).

So, I have attempted to install my own version of Python instead, but quickly run into this problem:

```
make -j1 -C ports/python/virtualenv do-install
/com/extra/python/2.7/bin/python
make[1]: Entering directory `/home/terrymwh/bin/pitchfork/ports/python/virtualenv'
/com/extra/python/2.7/bin/python /home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py /home/terrymwh/bin/pitchfork/deployment
Traceback (most recent call last):
  File "/home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py", line 14, in <module>
    import shutil
  File "/com/extra/python/2.7/lib/python2.7/shutil.py", line 12, in <module>
    import collections
  File "/com/extra/python/2.7/lib/python2.7/collections.py", line 9, in <module>
    from operator import itemgetter as _itemgetter
ImportError: /com/extra/python/2.7/lib/python2.7/lib-dynload/operator.so: undefined symbol: _PyUnicodeUCS2_AsDefaultEncodedString
make[1]: *** [/home/terrymwh/bin/pitchfork/deployment/var/pkg/virtualenv-13.1.2] Error 1
make[1]: Leaving directory `/home/terrymwh/bin/pitchfork/ports/python/virtualenv'
make: *** [python] Error 2
```

...and when I attempt to use the Anaconda package that was already installed on the cluster, it complains about too many symlinks:

```
make -j1 -C ports/python/virtualenv do-install
/com/extra/Anaconda-Python/2.2.0-2.7/bin/python
make[1]: Entering directory `/home/terrymwh/bin/pitchfork/ports/python/virtualenv'
/com/extra/Anaconda-Python/2.2.0-2.7/bin/python /home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py /home/terrymwh/bin/pitchfork/deployment
New python executable in /home/terrymwh/bin/pitchfork/deployment/bin/python
Traceback (most recent call last):
  File "/home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py", line 2363, in <module>
    main()
  File "/home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py", line 832, in main
    symlink=options.symlink)
  File "/home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py", line 994, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/home/terrymwh/bin/pitchfork/workspace/virtualenv-13.1.2/virtualenv.py", line 1288, in install_python
    shutil.copyfile(executable, py_executable)
  File "/com/extra/Anaconda-Python/2.2.0-2.7/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 40] Too many levels of symbolic links: '/home/terrymwh/bin/pitchfork/deployment/bin/python'
make[1]: *** [/home/terrymwh/bin/pitchfork/deployment/var/pkg/virtualenv-13.1.2] Error 1
make[1]: Leaving directory `/home/terrymwh/bin/pitchfork/ports/python/virtualenv'
make: *** [python] Error 2
```

Is there a workaround for this issue? My worries are that even if I manage to get the admins/maintainers of the cluster to install PySam in Anaconda, it will still run into a symlinks error.

**Comment by terrymun** [comment]
Alright, I have narrowed the issue down to the `htslib/hfile_libcurl.c` file. Pitchfork complains that:

```
    htslib/hfile_libcurl.c: In function ‘easy_errno’:
    htslib/hfile_libcurl.c:93:10: error: ‘CURLE_NOT_BUILT_IN’ undeclared (first use in this function)
         case CURLE_NOT_BUILT_IN:
              ^
    htslib/hfile_libcurl.c:93:10: note: each undeclared identifier is reported only once for each function it appears in
    error: command 'gcc' failed with exit status 1
```

Is it possible to force pitchfork to use an earlier version of pysam, i.e. ≤0.8.3?

**Comment by mhsieh** [comment]
I don't think mixing anaconda with pitchfork is supported at this moment, the unicode error makes me suspect that you might have some environment variables conflicting with pitchfork, can you post the result of this command?
```
env | grep -i python
```
As to the CURL_NOT_BUILT_IN problem, is it possible to install libcurl-dev or libcurl-devel in your system. What OS/Linux-distro are you using?

**Comment by mjhsieh** [comment]
Your system probably has an old curl binary that doesn't work well with https, and the libcurl-devel is probably missing. The pysam problem should be now fixed in the case of missing libcurl-dev. Fetching should be also fixed in recent commit from @evolvedmicrobe. I will close this first, if you still have problem building it, reopen this or file a new issue please. :)

**Comment by terrymun** [comment]
Sorry for the slow replies, and thanks for the very helpful insights. I have managed to circumvent the issue by actually running the commands in the SMRTanalysis shell that came with the SMRTanalysis package instead :)
